caffe2/share/contrib/binaries/caffe2_benchmark/CMakeLists.txt

```diff
@@ -1,6 +1,6 @@
 set(__target "caffe2_benchmark")
 add_executable(${__target} "${CMAKE_CURRENT_SOURCE_DIR}/caffe2_benchmark.cc")
 add_dependencies(${__target} ${Caffe2_MAIN_LIBS_ORDER})
-caffe_add_whole_archive_flag(Caffe2_CPU_OBSERVER Caffe2_CPU_OBSERVER_LINK)
+caffe_add_linker_flag(Caffe2_CPU_OBSERVER Caffe2_CPU_OBSERVER_LINK)
```

Comment:
does add_linker_flag preserve whole_archive?

Reply:
I will use -DBUILD_SHARED_LIBS=OFF so that only static library is built for benchmarking purpose. Otherwise it uses dynamic library, which is fine in most cases.